components/NfcShare.vue

```diff
@@ -0,0 +1,136 @@
+<template>
+  <div v-if="supportsNfc">
+    <button
+      v-if="!isSharing"
+      @click="writeNfc">
+      Share via NFC
+    </button>
+    <div v-if="isSharing">
+      Hold your device near a reader to share your credential.
+      <button @click="cancelWrite">
+        Cancel
+      </button>
+    </div>
+  </div>
+</template>
+
+<script>
+import {defineComponent, ref} from 'vue';
+import {Notify} from 'quasar';
+
+export default defineComponent({
+  props: {
+    credential: {
+      type: Object,
+      required: true
+    }
+  },
+  setup(props) {
+    const lenMaxNfcBytes = 32768;
+
+    const supportsNfc = ref(false);
+    const isSharing = ref(false);
+
+    let abortController = null;
+    let ndef = null;
+
+    if('NDEFReader' in window) {
+      supportsNfc.value = true;
+      // eslint-disable-next-line no-undef
+      ndef = new NDEFReader();
+    }
+
+    function cancelWrite() {
+      if(abortController) {
+        abortController.abort();
+      }
+      isSharing.value = false;
+    }
+
+    function notifyError(message) {
+      Notify.create({
```

Comment:
The other parts of this codebase use `$q.notify` and pass in a type of `positive` or `negative` mimicking that practice could add consistency across components.